modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/FieldSelectModalContent.tsx

```diff
@@ -203,7 +213,9 @@ const FieldSelectModalContent = ({
 				setFields(updatedFields);
 			}
 		});
-	}, [selectedFields, fdsView]);
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
```

Comment:
Before this change, we would be making two requests on modal open and two on modal close, along with react error. This is a quick & dirty fix. A much better solution could be to fetch fields only once for the entire view, but that can be a separate task.